src/cgutils.cpp

```diff
@@ -1397,7 +1397,13 @@ static Value *emit_getfield_knownidx(Value *strct, unsigned idx, jl_datatype_t *
         }
     }
     else if (strct->getType()->isPointerTy()) { // something stack allocated
+#       ifdef LLVM37
+        Value *addr = builder.CreateConstInBoundsGEP2_32(
+            cast<PointerType>(strct->getType()->getScalarType())->getElementType(),
+            strct, 0, idx);
```

Comment:
i think you can write the same a bit more concisely as `strct->getContainedType(0)`. i'm surprised as well that it's not the same as `elty` in some or all cases though.